### ISSUE TITLE
 Glossref is not really working as advertised

### ISSUE THREAD

**Issue by easirois** [dita-ot/dita-ot#2287]
It appears that glossref is not really working as advertised. All the docs say I should be able to have

bookmap
  chapter --> mapA.ditamap --> topicB.xml
  backmatter
      glossarylist --> glossary.ditamap --> <glossref href="term1.xml" keys="term1" />

and then if topicB.xml includes <abbreviated-form keyref="term1"/> I should see it resolved as a hyperlink in output to the corresponding term in my glossary. I do not.

Instead, it's necessary to structure glossary.ditamap like this:

<topicref href="term1.xml"/>
<keydef href="term1.xml" keys="term1" />

In other words, the OT does not interpret the glossref as a key definition, as the docs seem to say it should. I have to reference the glossentry topic as a topicref and then define a key to it.

**Comment by easirois** [comment]
Another way to make it work is to use <topicref> instead of <glossref>:

bookmap
  chapter --> mapA.ditamap --> topicB.xml
  backmatter
      glossarylist --> glossary.ditamap --> <topicref href="term1.xml" keys="term1" />

It seems to work by adding @toc=”yes” and/or @print=”yes” to <glossref>. (Which basically makes it a <topicref>…)

IMO: I think it should work with a glossref as well as well as the 'workaround'.. That said, is the issue here because glossref has a default value of toc/print set to no by default?

**Comment by robander** [comment]
Interesting one ... if going to PDF, and you've got something that defaults to `print="no"`, then I'm not sure there could be a target for links because the target will not exist in the PDF. For HTML it's less clear - you could create a link target to that file, it's just not going to show up in your TOC.

**Comment by easirois** [comment]
hmm...there seems to be a discussion about this issue in https://github.com/dita-ot/dita-ot/issues/1721
